HelperFunctions/ThirdParty/questdlg_nonmodal.m

```diff
@@ -0,0 +1,602 @@
+    
+elseif NumButtons==3
+    BtnXOffset(2)=(FigPos(3)-BtnWidth)/2;
+    BtnXOffset=[BtnXOffset(2)-DefOffset-BtnWidth
+        BtnXOffset(2)
+        BtnXOffset(2)+BtnWidth+DefOffset
+        ];
+end
+
+set(QuestFig ,'Position',getnicedialoglocation(FigPos, get(QuestFig,'Units')));
+assert(iscell(BtnHandle));
+BtnPos=cellfun(@(bh)get(bh,'Position'), BtnHandle, 'UniformOutput', false);
+BtnPos=cat(1,BtnPos{:});
+BtnPos(:,1)=BtnXOffset;
+BtnPos=num2cell(BtnPos,2);
+
+assert(iscell(BtnPos));
+cellfun(@(bh,pos)set(bh, 'Position', pos), BtnHandle, BtnPos, 'UniformOutput', false);
+
+if DefaultValid
+    setdefaultbutton(QuestFig, BtnHandle{DefaultButton});
+end
+
+delete(MsgHandle);
+
+
+set(texthandle, 'Position',[MsgTxtXOffset MsgTxtYOffset 0]);
+
+
+IconAxes=axes(                                      ...
+    'Parent'      ,QuestFig              , ...
+    'Units'       ,'Pixels'              , ...
+    'Position'    ,[IconXOffset IconYOffset IconWidth IconHeight], ...
+    'NextPlot'    ,'replace'             , ...
+    'Tag'         ,'IconAxes'              ...
+    );
+
+set(QuestFig ,'NextPlot','add');
+
+[iconData, alphaData] = matlab.ui.internal.dialog.DialogUtils.imreadDefaultIcon('quest');  
+Img=image('CData',iconData, 'AlphaData', alphaData, 'Parent',IconAxes);
+set(IconAxes, ...
+    'Visible','off'           , ...
+    'YDir'   ,'reverse'       , ...
+    'XLim'   ,get(Img,'XData')+[-0.5 0.5], ...
+    'YLim'   ,get(Img,'YData')+[-0.5 0.5]  ...
+    );
+
+% make sure we are on screen
+movegui(QuestFig)
+
+
+set(QuestFig ,'Visible','on');
+drawnow;
+
+if DefaultButton ~= 0
+    uicontrol(BtnHandle{DefaultButton});
+end
+
+if ishghandle(QuestFig)
+    % Go into uiwait if the figure handle is still valid.
+    % This is mostly the case during regular use.
+    c = matlab.ui.internal.dialog.DialogUtils.disableAllWindowsSafely();
+    uiwait(QuestFig);
+	delete(c);
+end
+
+% Check handle validity again since we may be out of uiwait because the
+% figure was deleted.
+if ishghandle(QuestFig)
+    if DefaultWasPressed
+        ButtonName=Default;
+    else
+        ButtonName = getappdata(get(QuestFig,'CurrentObject'),'QuestDlgReturnName');
+    end
+    doDelete;
+else
+    ButtonName='';
+end
+drawnow; % Update the view to remove the closed figure (g1031998)
+
+    function doFigureKeyPress(obj, evd)  %#ok
+    switch(evd.Key)
+        case {'return','space'}
+            if DefaultValid
+                DefaultWasPressed = true;
+                uiresume(gcbf);
+            end
+        case 'escape'
+            doDelete
+    end
+    end
+
+    function doControlKeyPress(obj, evd)  %#ok
+    switch(evd.Key)
+        case {'return'}
+            if DefaultValid
+                DefaultWasPressed = true;
+                uiresume(gcbf);
+            end
+        case 'escape'
+            doDelete
+    end
+    end
+
+    function doDelete(varargin)
+    delete(QuestFig);
+    end
+end
+
+function outStr = dialogCellstrHelper (inputStr)
```

Comment:
From here on, the source of additional private functions that `questdlg` needed are added as local functions to keep same behavior.